### ISSUE TITLE
Possible Race Condition When Establishing Pool Connections

### ISSUE THREAD

**Issue by cyounkins** [neumino/rethinkdbdash#199]
I'm seeing my trivial script hang, possibly due to a race condition.

Script source: https://gist.github.com/cyounkins/82a5de715196c62992e0

Output of script, first when it terminates correctly, then when it hangs.

```
Creating a pool connected to localhost:28015
Creating a pool connected to 192.168.198.13:28015
{ deleted: 0,
  errors: 0,
  inserted: 0,
  replaced: 0,
  skipped: 0,
  unchanged: 0 }
stopping because 'tables' array is empty
Draining the pool connected to 192.168.130.122:28015
Draining the pool connected to 192.168.198.13:28015
Fail to retrieve a second copy of server_status
--------------------------
Creating a pool connected to localhost:28015
{ deleted: 0,
  errors: 0,
  inserted: 0,
  replaced: 0,
  skipped: 0,
  unchanged: 0 }
stopping because 'tables' array is empty
Draining the pool connected to localhost:28015
Creating a pool connected to 192.168.198.13:28015
```

Any thoughts?

**Comment by cyounkins** [comment]
Forgot some details:

rethinkdbdash version 2.2.8
two servers running rethinkdb 2.2.1
Node v0.10.24

**Comment by neumino** [comment]
Thanks for the report. Fixed in 2.2.9

**Comment by cyounkins** [comment]
Still seeing this in 2.2.9 :-(

**Comment by neumino** [comment]
What's the output of the script? I can't reproduce it.

**Comment by cyounkins** [comment]
I'm seeing the exact same output as in the original post. Could "Fail to retrieve a second copy of server_status" be related?

**Comment by neumino** [comment]
I can reproduce if I set a timeout of 3ms on the drain. There's a fix in 2.2.10.
Let me know if it still happens with the last version.

**Comment by cyounkins** [comment]
That appears to have fixed it. Thank you!


Craig Younkins
